drivers/properties/serial/zephyr,native-posix-uart.yaml

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022 Trackunit A/S
```

Comment:
Seems like we should have `uart.yaml` specify `api: uart` and include `uart.yaml` in driver specific YAMLs

Reply:
Would also like to see how we specify things like standard uart features, like does this driver support async or not.

Reply:
The perfect way of doing this would be to have multiple APIs for it
`api: ["uart_poll", "uart_async", "uart_interrupt"]`

bit since that is not possible right now, the uart.yaml would contain:
```
properties:
  supports_uart_poll:
    type: bool
  supports_uart_async:
    type: bool
  supports_uart_interrupt:
    type: bool 
```

The specific uart file would then look like this:
```
include: "uart.yaml"

compatible: "dummy,uart"

properties:
  supports_uart_poll:
    value: true
```

This is just an example, we can easily add new keys and features to the files if we find a better way of doing it

Reply:
I have a better idea:

Lets add a key named features:

uart.yaml:
```
features:
    define:
      - uart_poll
      - uart_interrupt
      - uart_async
```

and dummy,uart.yaml
```
include: "uart.yaml"

compatible: "dummy,uart"

features:
  implements:
    - uart_poll
    - uart_interrupt
```

which could be tested for using
```
DEVICE_DT_FEATURE_SUPPORTED(node_id, uart_poll)
```

Reply:
Better idea yet

uart.yaml:
```
api:
  uart:
    features:
      - poll
      - interrupt
      - async
```

and dummy,uart.yaml
```
include: "uart.yaml"

compatible: "dummy,uart"

api:
  uart:
    implements:
      - poll
      - interrupt
```

and to test for features
```
DEVICE_DT_API_IMPLEMENTS(node_id, uart, poll)
```

Reply:
> The perfect way of doing this would be to have multiple APIs for it `api: ["uart_poll", "uart_async", "uart_interrupt"]`
> 
> bit since that is not possible right now, the uart.yaml would contain:
> 
> ```
> properties:
>   supports_uart_poll:
>     type: bool
>     required: true
>   supports_uart_async:
>     type: bool
>     required: true
>   supports_uart_interrupt:
>     type: bool
>     required: true
> ```
> 
> The specific uart file would then look like this:
> 
> ```
> include: "uart.yaml"
> 
> compatible: "dummy,uart"
> 
> properties:
>   supports_uart_poll:
>     value: true
>   supports_uart_interrupt:
>     value: true
>   supports_uart_async:
>     value: false
> ```
> 
> Which can be determined at compile time using
> 
> ```
> #if DEVICE_DT_PROPERTY(node_id, supports_uart_interrupt)
> #endif
> ```
> 
> **Note!** that they properties dont support booleans yet, i will add this shortly, I forgot about that one :)

We have Kconfig to specify wether a serial driver supports async/interrupt: `SERIAL_SUPPORT_ASYNC`, `SERIAL_SUPPORT_INTERRUPT`. Why do we need them in another file now?